dbschema/project.esdl

```diff
@@ -119,6 +119,41 @@ module default {
         projectContext := __new__.projectContext,
       }
     );
+
+    trigger createPeriodicReportsOnInsert after insert for each do (
+      with
+        interval := (select 
+          if __new__.financialReportPeriod = default::ReportPeriod.Monthly then '1' else '3'),
+        reportRanges := Project::create_periodic_report_ranges(
+          __new__.mouStart,
+          __new__.mouEnd,
+          interval
+        )
+      for reportRange in reportRanges 
+      union (
+        (insert default::FinancialReport {
+          createdAt := datetime_of_statement(),
+          modifiedAt := datetime_of_statement(),
+          createdBy := assert_exists(global currentActor),
+          modifiedBy := assert_exists(global currentActor),
+          project := __new__,
+          projectContext := __new__.projectContext,
+          container := __new__,
+          period := reportRange
+          #receivedDate := __new__.financialReportReceivedAt, //TODO - what is this at the project level?
```

Comment:
That's what the user defines when the file is uploaded. "Here is the word doc from the field partner which I'm uploading on the 7th, but I actually received the email from them on the 4th"